### PR TITLE
feat: Spotify segment for windows(no dependency)

### PR DESCRIPTION
**Pull request by lnu**
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

Windows support added for spotify segment. AutoHotkey is used to get the informations. AutoHoykey script added to the project. Should be deployed in the same folder as the oh-my-posh executable.

Preview:
ohmyposh_spotify_windows

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by lnu** [comment]
The win32 interface has been created but I don't know exactly where to inject it. Right now I added a function in environment.go with a windows implementation and a not implemented error for others.

File: win32.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"github.com/mitchellh/go-ps"
```
Comment (JanDeDobbeleer) [inline win32.go]: We removed this dependency in favor of [gopsutil](https://github.com/shirou/gopsutil). Can you double check if we can also use `gopsutil` here?

File: win32.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+)
+
+type win32Info interface {
```
Comment (JanDeDobbeleer) [inline win32.go]: As this is now part of `environment`, the interface here is redundant. I can live with it being removed as mocking is possible via `MockedEnviroment` anyhow.

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

**Comment by JanDeDobbeleer** [comment]
Almost there now! 👏🏻

File: win32.go (hunk 1 above)
Comment (lnu) [reply win32.go]: I checked but there's a difference in the function that returns the processes.  
`go-ps` executes only one call to CreateToolhelp32Snapshot and then will loop through the snapshot with Process32First and Process32Next, which is very fast.
`gopsutil` executes a call to get the process pids, but when we acces the process name, it executes a call to  CreateToolhelp32Snapshot  for each process, which is very slow.
I'm still investigating there's a way to have the same behaviour with gopsutil.

Comment (JanDeDobbeleer) [reply win32.go]: That would be a valid reason to use `go-ps` in that case.

Comment (lnu) [reply win32.go]: dependency removed and only needed code added

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]
One last thing and we can merge this, can you squash all of these in a single commit? No need to see separate history for these, we can treat this as 1.

**Comment by lnu** [comment]
Squash and rebase. I’ll do it tonight.

**Pull request merged**